packages/bpk-component-popover/src/BpkPopover-test.tsx

```diff
@@ -191,4 +191,41 @@ describe('BpkPopover', () => {
       expect(onCloseSpy).toHaveBeenCalled();
     });
   });
+
+  it('should call target onClick handler and execute code after stopPropagation', () => {
+    const handleClick = jest.fn((event) => {
+      event.stopPropagation();
+      // eslint-disable-next-line no-param-reassign
+      event.afterStopPropagation = true;
+    });
+
+    const { getByRole } = render(
+      <BpkPopover
+        id="my-popover"
+        onClose={jest.fn()}
+        label="My popover"
+        closeButtonLabel="Close"
+        labelAsTitle
+        closeButtonIcon
+        target={<button onClick={handleClick} type="button">My target</button>}
+        isOpen
+      >
+        My popover content
+      </BpkPopover>,
+    );
+
+    const button = getByRole('button', { name: 'My target' });
+
+    const event = new MouseEvent('click', { bubbles: true });
+    jest.spyOn(event, 'stopPropagation');
+
+    act(() => {
+      fireEvent(button, event);
+    });
+
+    expect(handleClick).toHaveBeenCalled();
+    expect(event.stopPropagation).toHaveBeenCalled();
+    expect(handleClick.mock.calls[0][0].afterStopPropagation).toBe(true);
```

Comment:
To avoid the error, change this to similar as above to be

```suggestion
    await waitFor(async () => {
      await fireEvent(button, event);
      expect(handleClick).toHaveBeenCalled();
      expect(event.stopPropagation).toHaveBeenCalled();
      expect(handleClick.mock.calls[0][0].afterStopPropagation).toBe(true);
    });
```